strax/storage/mongo.py

```diff
@@ -208,6 +214,11 @@ def _save_chunk(self, data, chunk_info, executor=None):
         # for this chunk
         chunk_id = self.ids_chunk.get(chunk_i, None)
         if chunk_id is not None:
+            # We can fail here if the document is too large to be
+            # written out to mongo. One could do a try:
+            # except pymongo.errors.WriteError: pass
+            # but that potentially leads to abuse of a Mongo instance
+            # going unnoticed.
             self.col.update_one({'_id': chunk_id},
                                 {'$addToSet': {f'data': aggregate_data}})
```

Comment:
This probably isn't the best place for this discussion, but I've been thinking about our use here of `$addToSet`. This is a relatively "heavy" operator, because it has to check every other element in the list for equality. For cases where we're saving a histogram, there are a lot of elements that need to be checked. I assume it optimizes, but if the two histograms are equal, it's still a _lot_ of operations. I don't remember the details of when Savers get invoked (middle of a chunk? End of a chunk?), but the performance would probably increase if we can find a different operator to use.

Reply:
Perhaps https://docs.mongodb.com/manual/reference/operator/update/push/ is a better operator looking at the documentation now. There is no need to indeed double check that the data is not stored yet.

Also, in principle I'm actually expecting many documents to be updated here but it depends a little on the re-chunking of documents (disabled by default). In principle each document is just uploaded as soon as the input-chunk has been seen. This means that the $addToSet would not really be used as long as people don't start modifying this default. 

Nonetheless it does not need to check it's written twice so let me just change it to $push

Reply:
Rechunking is a tricky subject in the context of the Mongo interface, I think we can make a coherent argument for prohibiting it for this saver. If users are overriding `allow_rechunk=False` for this saver then they probably know what they're doing (or we can at least hope).